find_changed_files.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+# Find changed files in a build either via pull request API or compare API
+echo "event path: $GITHUB_EVENT_PATH"
+cat $GITHUB_EVENT_PATH
+COMPARE=$(jq '.compare' $GITHUB_EVENT_PATH)
+PR=$(jq -r '.pull_request._links.self.href' $GITHUB_EVENT_PATH)
+echo "-----"
+echo "compare: |$COMPARE|"
+echo "PR: |$PR|"
+echo "------"
+if [[ $COMPARE != null ]]
```

Comment:
Is this the path regular push goes through? What other event types are supported? Can we leave a comment for that here.

What happens if you push a multiple commits / they get batched by copybara?

Reply:
honestly idk about what would happen w/ copybara.

for PR case, we have multiple commits and it works.

for pushing regular commits, i keep getting only 1 commit cases so idk how i can test what copybara does.
That being said, as long as the compare URL is there, it should work because it is in the root object (not specific to an individual commit)

Reply:
https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push

push event can have multiple commits here:

actually it looks like the compare link does a range of commits so we're okay here.

`https://github.com/Codertocat/Hello-World/compare/6113728f27ae...000000000000`